core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -694,7 +694,14 @@ public static RangeSet fromString(String list, boolean skipError) {
                 try {
                     if(s.contains("-")) {
                         String[] tokens = Util.tokenize(s,"-");
-                        rs.ranges.add(new Range(Integer.parseInt(tokens[0]),Integer.parseInt(tokens[1])+1));
+                        if (tokens.length == 2) {
+                            rs.ranges.add(new Range(Integer.parseInt(tokens[0]), Integer.parseInt(tokens[1]) + 1));
+                        }
+                        else {
+                            if (!skipError)
+                                throw new IllegalArgumentException("Unable to parse " + list);
```

Comment:
NIT: missing brackets. It would be also useful to explain the formatting requirements in the exception. E.g. "expected string with format XX-YY" or something like that. Maybe Int parsing errors should be caught & processed as well